ctapipe/calib/camera/calibrator.py

```diff
@@ -218,18 +218,27 @@ def _calibrate_dl1(self, event, telid):
             if self.image_extractor.requires_neighbors():
                 camera = event.inst.subarray.tel[telid].camera
                 self.image_extractor.neighbors = camera.neighbor_matrix_where
+            # TODO: apply timing correction to waveforms before charge extraction
             charge, pulse_time = self.image_extractor(waveforms)
 
             # Apply integration correction
             # TODO: Remove integration correction
             correction = self._get_correction(event, telid)
             corrected_charge = charge * correction
 
-        event.dl1.tel[telid].image = corrected_charge
+        # Calibrate extracted charge
+        try:
+            pedestal = event.mon.tel[telid].dl1.pedestal
+            absolute = event.mon.tel[telid].dl1.absolute
+            relative = event.mon.tel[telid].dl1.relative
+            calibrated_charge = (corrected_charge - pedestal) * relative / absolute
```

Comment:
Why have relative and absolute and not a single constant here?

Reply:
Traditionally in HESS the calibration coefficient is split into two. I think this is due to the separate techniques used to derive each:
- gain: extracted from SPE
- flat-fielding: extracted from uniform bright illumination to correct for the relative Quantum Efficiencies between PMTs

If each parameter is derived from a separate source (separate dataset, separate script), then perhaps its logical to keep this distinction. But I suppose that is a question to the Camera Calibration coordinator...

Reply:
I assume *absolute* is the DC/PE measurement, which is computed once every so often (and loaded from a Service table), whereas the relative is the FlatFielding correction, which is computed either on-the-fly, or in a previous pass over the same event data (looking at flasher events).  So the two are computed in different places.   Perhaps naming them less generally would help this?

Reply:
Note that you can still for example do reconstruction, without the absolute gain correction, relying only on the relative correction, but then it makes comparing different cameras more complicated (since any cuts on intensity will be different for each camera).   In principle you can also get away with no relative correction (flatfield) if you use a model analysis or something that doesn't rely on the image having a uniform response, but then computing things like Hillas parameters is annoying.

Reply:
The "general naming" was on purpose. 

The different equivalent names for the absolute calibration are DC2PE and gain. But these can be misnomers when calibrating SiPMs. As the QE (referred in SiPMs as PDE) also changes with the voltage across the photosensor, it is more appropriate to calibrate from DC (or mV) to photons, to ensure the same illumination response is present in each SST.

The other name for relative as you said is flat-fielding. I am not opposed as much against that name. Just "absolute" and "relative" seemed more descriptive about what they are calculated against.

Reply:
Ok agreed, general names are better (but still not clear if they are scales or shifts)